metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1087,9 +1111,15 @@ void handleBrokerFenced(int brokerId, List<ApiMessageAndVersion> records) {
         }
         generateLeaderAndIsrUpdates("handleBrokerFenced", brokerId, NO_LEADER, records,
             brokersToIsrs.partitionsWithBrokerInIsr(brokerId));
-        records.add(new ApiMessageAndVersion(new FenceBrokerRecord().
-            setId(brokerId).setEpoch(brokerRegistration.epoch()),
-            FENCE_BROKER_RECORD.highestSupportedVersion()));
+        if (featureControl.metadataVersion().brokerRegistrationChangeRecordSupported()) {
+            records.add(new ApiMessageAndVersion(new BrokerRegistrationChangeRecord().
+                    setBrokerId(brokerId).setBrokerEpoch(brokerRegistration.epoch()).setFenced((byte) 1),
+                    (short) 0));
```

Comment:
How about using `BrokerRegistrationChangeRecord.highestSupportedVersion()`  instead of using fixed value here.

Reply:
We want the fixed value, since any change in emitted version requires a change in MetadataVersion.